### PR TITLE
MOODLE-5778 - Added default gradetype and gradescale

### PR DESCRIPTION
**Pull request by frederikmillingpytlick**

### CODE REVIEW

File: course/moodleform_mod.php [hunk 1]
```diff
@@ -1024,6 +1025,20 @@ public function standard_grading_coursemodule_elements() {
...
+                }
+
+                // if the gradetype is GRADE_TYPE_SCALE and a default gradescale is set, then pass the default gradescale through to the modgrade group element
+                if ((int)$mod_gradetype_config === GRADE_TYPE_SCALE) {
+                    $mod_gradescale_config = get_config("mod_{$this->_modname}", "gradescale");
+                    if (!$mod_gradescale_config === false) {
+                        $gradeoptions['currentscaleid'] = (int)$mod_gradescale_config;
+                    }
+                }
             }
```
Comment (johnnypraxis) [inline course/moodleform_mod.php:1042]: Please avoid all these if else statements. Replace them with small methods with meaningful names.

File: lib/form/modgrade.php [hunk 2]
```diff
@@ -514,6 +514,22 @@ public function onQuickFormEvent($event, $arg, &$caller) {
...
+                                $this->scaleformelement->setValue($this->currentscaleid);
+                                break;
+                            case 'point':
+                                $this->maxgradeformelement->setValue(100);
+                                break;
+                        }
+
+                        break;
+                    }
+
```
Comment (johnnypraxis) [inline lib/form/modgrade.php:532]: Same as above. Extract.

**Review by johnnypraxis — COMMENTED** [review]
See comments.

**Review by johnnypraxis — APPROVED** [review]

**Comment by frederikmillingpytlick** [comment]
Closing, this feature will not be coming before Moodle 4.2+

**Pull request closed without merging**